mod_prometheus.rs

```diff
@@ -316,36 +316,68 @@ fn prometheus_load(mod_int: &ModInterface) -> Status {
                 if let Some(billsecvar) = e.header("variable_billsec") {
-                    let parsed_time = billsecvar.parse::<i64>();
+                    let parsed_time = billsecvar.parse::<u64>();
                     if parsed_time.is_ok() {
                         let bill_seconds = parsed_time.unwrap() ;
-                        if let Some(direction) = e.header("Call-Direction") {
-                            if direction == "outbound" {
-                                COUNTERS[FSCounter::SessionsOutboundCallDurationTotal].lock().unwrap().increment_by(bill_seconds as f64);
-                                COUNTERS[FSCounter::SessionsOutboundCallHangupComplete].lock().unwrap().increment();
-                                let acd = COUNTERS[FSCounter::SessionsOutboundCallDurationTotal].lock().unwrap().value() /
-                                                                COUNTERS[FSCounter::SessionsOutboundCallHangupComplete].lock().unwrap().value();
-                                GAUGES[FSGauge::SessionsOutboundACD].lock().unwrap().set(acd);
-                            } else if direction == "inbound" {
-                                COUNTERS[FSCounter::SessionsInboundCallDurationTotal].lock().unwrap().increment_by(bill_seconds as f64);
-                                COUNTERS[FSCounter::SessionsInboundCallHangupComplete].lock().unwrap().increment();
-                                let acd = COUNTERS[FSCounter::SessionsInboundCallDurationTotal].lock().unwrap().value() /
-                                                COUNTERS[FSCounter::SessionsInboundCallHangupComplete].lock().unwrap().value();
-                                GAUGES[FSGauge::SessionsInboundACD].lock().unwrap().set(acd);
-                            }
+                        if direction == "outbound" {
+                            COUNTERS[FSCounter::SessionsOutboundCallDurationTotal].lock().unwrap().increment_by(bill_seconds as f64);
+                            COUNTERS[FSCounter::SessionsOutboundCallHangupComplete].lock().unwrap().increment();
+
+                            let totalSeconds: u64 = COUNTERS[FSCounter::SessionsOutboundCallDurationTotal].lock().unwrap().value() as u64;
+                            let totalHup: u64 = COUNTERS[FSCounter::SessionsOutboundCallHangupComplete].lock().unwrap().value() as u64;
+                            let acd_out = totalSeconds / totalHup;
+
+                            GAUGES[FSGauge::SessionsOutboundACD].lock().unwrap().set(acd_out as f64);
+
+                            fslog!(INFO, "callid:{:#?} uniqueId:{:#?} {:#?} bill:{:#?} sec. totalHup:{:#?} total:{:#?} sec. acd:{:#?} \n",
+                                callid, uniqueId, direction, bill_seconds, totalHup, totalSeconds, acd_out);
+
+                        } else if direction == "inbound" {
+
+                            COUNTERS[FSCounter::SessionsInboundCallDurationTotal].lock().unwrap().increment_by(bill_seconds as f64);
```

Comment:
shouldn't it be u64 as well?

Reply:
no, this is how the function has been designed by Moises Sylva